### PR TITLE
Reorg the RemoteVMManager

### PR DESCRIPTION
**Pull request by dzhengfy**
The class RemoteVMManager is not suitable in the __init__.py.  All
remote utilities are decided to be put into remote.py as a central place.
Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Review by sathnaga — CHANGES_REQUESTED** [review]
@dzhengfy agree, it is good to be placed out outside utils_test/__init__.py, i feel it is good candidate for `virttest/remote.py` what say? as most relevant methods are in place already...

**Comment by sathnaga** [comment]
@dzhengfy additionally, name suggest for class `VMManager` so it can be used as `remote.VMManager`

**Comment by dzhengfy** [comment]
> @dzhengfy additionally, name suggest for class `VMManager` so it can be used as `remote.VMManager`

@sathnaga Thanks for your suggestion. But it is not easy to detect the class is targeted for managing a VM on remote host with the name 'VMManager'. What do you think?

**Comment by dzhengfy** [comment]
> @dzhengfy agree, it is good to be placed out outside utils_test/**init**.py, i feel it is good candidate for `virttest/remote.py` what say? as most relevant methods are in place already...

@sathnaga Thanks for your suggestion.
I noticed 'remote.py' does already include some basic exception classes and some general utility functions as well as some classes. From my POV, I hope this py only includes some basic functions in infrastructure level, like remote_login(), scp_to_remote. And the classes which includes higher level functions which may invoke the basic functions in remote.py are better to be put in another py, like utils_remote.py I created. I am trying to make the file structure cleaner. 

This is just my thoughts. And welcome your comments to discuss. Thank you.

**Comment by sathnaga** [comment]
> > @dzhengfy agree, it is good to be placed out outside utils_test/**init**.py, i feel it is good candidate for `virttest/remote.py` what say? as most relevant methods are in place already...
> 
> @sathnaga Thanks for your suggestion.
> I noticed 'remote.py' does already include some basic exception classes and some general utility functions as well as some classes. From my POV, I hope this py only includes some basic functions in infrastructure level, like remote_login(), scp_to_remote. And the classes which includes higher level functions which may invoke the basic functions in remote.py are better to be put in another py, like utils_remote.py I created. I am trying to make the file structure cleaner.
> 
> This is just my thoughts. And welcome your comments to discuss. Thank you.

@dzhengfy 
My idea is to contain all the remote related utils/class into a single file, so that is ease maintenance and usage better, I am fine the name of the file being `remote.py` or `utils_remote.py` but it is good to have this under one roof.

regarding the name of the class from `RemoteVMManager ` to `VMManager` , i felt the later will be understood that once the class in placed under `*remote.py`, it will be easily identifiable, but feel free to have it as is.

**Comment by dzhengfy** [comment]
> > > @dzhengfy agree, it is good to be placed out outside utils_test/**init**.py, i feel it is good candidate for `virttest/remote.py` what say? as most relevant methods are in place already...
> > 
> > 
> > @sathnaga Thanks for your suggestion.
> > I noticed 'remote.py' does already include some basic exception classes and some general utility functions as well as some classes. From my POV, I hope this py only includes some basic functions in infrastructure level, like remote_login(), scp_to_remote. And the classes which includes higher level functions which may invoke the basic functions in remote.py are better to be put in another py, like utils_remote.py I created. I am trying to make the file structure cleaner.
> > This is just my thoughts. And welcome your comments to discuss. Thank you.
> 
> @dzhengfy
> My idea is to contain all the remote related utils/class into a single file, so that is ease maintenance and usage better, I am fine the name of the file being `remote.py` or `utils_remote.py` but it is good to have this under one roof.
> 
> regarding the name of the class from `RemoteVMManager ` to `VMManager` , i felt the later will be understood that once the class in placed under `*remote.py`, it will be easily identifiable, but feel free to have it as is.

@sathnaga , I accept your suggestion and makes updates. Please help double review.

**Comment by dzhengfy** [comment]
Local test based on the change:

```
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.migrate_vm.positive_testing.migration_with_devices.attach_virtual_disk.with_postcopy
JOB ID     : 4419b80e5b9f15707f063aeaffd6078fecf9a017
JOB LOG    : /root/avocado/job-results/job-2020-09-01T10.34-4419b80/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.migration_with_devices.attach_virtual_disk.with_postcopy: PASS (161.03 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 164.38 s

# avocado run --vt-type libvirt --vt-machine-type q35 virsh.migrate_options_shared.positive_test.pty_channel.with_postcopy
JOB ID     : bba6e578e062c7c89e06e7c8340df7da4d375a32
JOB LOG    : /root/avocado/job-results/job-2020-09-01T10.15-bba6e57/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.pty_channel.with_postcopy: PASS (258.27 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 260.70 s
```

File: virttest/remote.py [hunk 1]
```diff
@@ -1351,3 +1351,114 @@ def run(self, command, timeout=60, ignore_status=False):
...
+            self.runner.run("ssh-keygen -t rsa -q -N '' -f %s" % pri_key)
+        else:
+            logging.info("SSH key pair already exist")
+        session = self.runner.session
+        # To avoid the host key checking
+        ssh_options = "%s %s" % ("-o UserKnownHostsFile=/dev/null",
+                                 "-o StrictHostKeyChecking=no")
+        session.sendline("ssh-copy-id %s -i %s root@%s"
+                         % (ssh_options, pub_key, vm_ip))
+        while True:
```
Comment (sathnaga) [inline virttest/remote.py:1389]: handle_prompts() method will be work for this below login handle right?

Comment (dzhengfy) [reply virttest/remote.py]: @santwana Originally I did not consider to change the logic of the codes, but only simply move it.  Now I do not care to do some simple changes. So I accept your suggestion and did updates and test one case using setup_ssh_auth().

**Comment by dzhengfy** [comment]
Retest:
```
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.migrate_options_shared.positive_test.pty_channel.with_postcopy
JOB ID     : cd1e8e452ef1a94bdfcd9990cf3710c26f139cdd
JOB LOG    : /root/avocado/job-results/job-2020-09-02T11.33-cd1e8e4/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.pty_channel.with_postcopy: PASS (267.01 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 269.67 s

Log:
2020-09-02 11:36:24,488 remote           L0212 DEBUG| Got shell prompt -- logged in
```

File: virttest/remote.py [hunk 2]
```diff
@@ -1351,3 +1354,91 @@ def run(self, command, timeout=60, ignore_status=False):
         if status and (not ignore_status):
             raise process.CmdError(command, cmd_result)
         return cmd_result
+
+
+class VMManager(object):
```
Comment (chunfuwen) [inline virttest/remote.py:1359]: it sounds like class RemoteVMManager name is more explicit than VMManager  according to below comment "Manage VM on remote host"

File: virttest/remote.py [hunk 3]
```diff
@@ -156,7 +156,8 @@ def handle_prompts(session, username, password, prompt, timeout=10,
                  r"[Pp]lease wait", r"[Ww]arning", r"[Ee]nter.*username",
                  r"[Ee]nter.*password", r"[Cc]onnection timed out", prompt,
                  r"Escape character is.*",
-                 r"Command>"],
+                 r"Command>",
+                 r"lost connection"],
```
Comment (chunfuwen) [inline virttest/remote.py:160]: Do we need consider capital letter for lost connection, e.g. [Ll]ost [Cc]onnection ?

Comment (dzhengfy) [reply virttest/remote.py]: @chunfuwen , ' [Cc]onnection' should not be necessary as it is after 'lost'. I updated with '[Ll]lost'. Thanks.

**Review by kylazhang — APPROVED** [review]

**Review by chunfuwen — APPROVED** [review]

File: virttest/remote.py (hunk 1 above)
Comment (dzhengfy) [reply virttest/remote.py]: @santwana, I am sorry that my colleague did not know  I was waiting for your reply and he merged the pr.  If you still have comments, feel free to add in this pr and I would like to consider in another pr. Thanks.

**Pull request merged**